### PR TITLE
Enable setting locale and timezone

### PR DESCRIPTION
**Pull request by slakkala**
## Description of changes

Re-enables locale setting in systemd-localed, with added support for LOCALE_ARCHIVE.

Removed hardcoded timezone setting to allow timedatectl set-timezone to work.

## Checklist for things done

- [x] Summary of the proposed changes in the PR description
- [x] More detailed description in the commit message(s)
- [x] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [x] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [ ] Test procedure described (or includes tests). Select one or more:
  - [x] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [x] Author has run `nix flake check --accept-flake-config` and it passes
- [x] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [x] Author has added reviewers and removed PR draft status

## Testing

Run: `/nix/store/*-cli/bin/givc-cli --addr admin-vm-debug --name admin-vm-debug --port 9001 --notls set-locale ar_AE.UTF-8` and start chromium; observe UI language in Arabic. Run again with `set-timezone Europe/Helsinki`, observe UI time change. Reboot device, and changes should persist.

### CODE REVIEW

File: modules/microvm/virtualization/microvm/appvm.nix [hunk 1]
```diff
@@ -116,6 +117,13 @@ let
               nixpkgs.buildPlatform.system = configHost.nixpkgs.buildPlatform.system;
               nixpkgs.hostPlatform.system = configHost.nixpkgs.hostPlatform.system;
 
+              environment.extraInit = ''
```
Comment (mbssrc) [inline modules/microvm/virtualization/microvm/appvm.nix:120]: Isn't this imported with common?

Comment (slakkala) [reply modules/microvm/virtualization/microvm/appvm.nix:120]: Yes, yes it is. Thanks, forgot to remove this.

File: modules/microvm/virtualization/microvm/guivm.nix [hunk 2]
```diff
@@ -130,6 +131,12 @@ let
               XDG_PICTURES_DIR = "$HOME/Pictures";
               XDG_VIDEOS_DIR = "$HOME/Videos";
             };
+            extraInit = ''
```
Comment (mbssrc) [inline modules/microvm/virtualization/microvm/guivm.nix:134]: This as well?

Comment (slakkala) [reply modules/microvm/virtualization/microvm/guivm.nix]: Indeed. Removed, thanks.

**Comment by johannarautanen** [comment]
Tried with lenovo X1. flash-script uised.


After giving the commands `/nix/store/*/bin/givc-cli --addr admin-vm-debug --name admin-vm-debug --port 9001 --notls set-locale` and` set-timezone Europe/Helsinki`, the chromium language changed to Arabic and timezone to Helsinki.

**Issues:**
-But, after rebooting the chromium language was English and time zone UTC.
-And after rebooting chromium and Trusted  Browser apps didn't opened during first tries, need to select them couple times before those apps opened.
-Wasn't able to create ssh connection to the admin-vm-debug.

**Comment by slakkala** [comment]
Seems that microvm optimization and storagevm do not play along nicely. Disabled optimization for admin-vm, and things seem to work again; also small update to testing instructions.

**Comment by johannarautanen** [comment]
**Tested with lenovo X1**. flash-script used.

Working: 
- After entering` /nix/store/*-cli/bin/givc-cli --addr admin-vm-debug --name admin-vm-debug --port 9001 --notls set-locale ar_AE.UTF-8` and `set-timezone Europe/Helsinki`, the chromium and Trusted browser language changed to Arabic and timezone changed to Helsinki.
- Changed language and timezone was still available after logout, reboot and shutdown
- able to create ssh connection to admin-vm-debug
- chromium and Trusted browser opened after first try after rebooting.
- changed timezone taken into use ie in slack, teams.
In Microsoft while creating the meeting, the timezone changing noticed and after that the new time taken to use in Microsoft365 and Outlook.

- all other ghaf apps can be launched
- ci-test automation cases ok


Notes:
- When you open the "Switch windows"-windows, part of the apps are showing "Arabic-way"
Screenshot from 2024-10-25 10-21-07

- In  "Trusted Browser" the "Open page in normal page"-text is not localized to Arabic
Screenshot from 2024-10-25 10-25-27

-PrintScreen app "satty's" up and down toolbar's change the directions after Arabic-language reboot, ie the "save" -button is now left side, not in right side
Screenshot from 2024-10-25 10-29-54


- when you open Microsoft 365, outlook and Teams-app, there appeared note in Arabic, that "This page could not be be translated"
Screenshot from 2024-10-25 10-16-08

**Review by brianmcgillion — APPROVED** [review]

**Pull request merged**